src/config.cc

```diff
@@ -516,6 +525,17 @@ Status Config::Load(const std::string &path) {
     std::cout << "Warn: no config file specified, using the default config. "
                     "In order to specify a config file use kvrocks -c /path/to/kvrocks.conf" << std::endl;
   }
+
+  // Validate again when read all the config items from the file.
+  for (const auto &iter : fields_) {
```

Comment:
I think it would confuse others why to validate again here. Also, this validation was needed when dynamic config the `masterauth` and `requirepass`.

After reconsidering this issue, maybe we can rename the `callback` function to postValidate and `validate` to preValidate if you think the `callback` name was NOT suitable to add this validation.

Reply:
I think `postValidate` is inferior to `callback`, so I'll put this check in `callback`

Reply:
yeah, what i mean was we also need to validate it when use `config set`. So put it on the callback also makes sense for me.

Reply:
The behavior of `CONFIG SET` is validate -> set -> callback. When `callback` is called, the configuration has been modified; If placed in `callback`, the configuration needs to be rollback.

Reply:
hmm... Yes

Reply:
So perhaps it would be better to save the line number when reading the configuration file and then validate it after reading all the configuration items?

Reply:
Yes, it should be better.